src/SmoFacade/AvailabilityGroup.cs

```diff
@@ -57,7 +57,13 @@ public void JoinPrimary(string dbName)
 
     public void Remove(string dbName)
     {
-      _availabilityGroup.AvailabilityDatabases[dbName]?.Drop();
+      if (IsPrimaryInstance)
+      {
+        _availabilityGroup.AvailabilityDatabases[dbName]?.Drop();
+      } else
+      {
+        _availabilityGroup.AvailabilityDatabases[dbName]?.LeaveAvailabilityGroup();
+      }
```

Comment:
I haven't looked into what the diff b/w the two calls are yet and what `HADR` means, but just wondering before I do...
It looks like, in the old code, we weren't removing the DB from the secondary AG before trying to drop the DB on the secondary servers
Could it just be that which was causing the errors?